### PR TITLE
{WIP}(PUP-8771) Add NTFS permission tests for Puppet settings

### PR DESCRIPTION
**Pull request by glennsarti**
DO NOT MERGE
TODO

- [ ] Do I need to worry about localised names for LOCAL SYSTEM or Administrators?
- [ ] Do I need to worry about admin v non-admin settings scenarios?

### CODE REVIEW

**Comment by glennsarti** [comment]
Pending test looks like
```
Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Puppet::Settings when applying the catalog generated from Puppet.settings should not add the NT AUTHORITY\SYSTEM account, but should add BUILTIN\Administrators
     # PUP-6729 should resolve this
     Failure/Error: end
       expected that {"InheritanceFlags"=>"ContainerInherit, ObjectInherit", "AccessControlType"=>"Allow", "IsInherited"=>false, "Path"=>"C:/Users/glenn.sarti/AppData/Local/Temp/rspecrun20180529-21804-upsfl6/tmp_settings20180529-21804-1liaocy/var/log", "FileSystemRights"=>"FullControl", "IdentityReference"=>"NT AUTHORITY\\SYSTEM", "PropagationFlags"=>"None"} would not contain an ACE for identity NT AUTHORITY\SYSTEM
     # ./spec/integration/util/settings_ntfs_spec.rb:142:in `block (5 levels) in <top (required)>'
     # ./spec/integration/util/settings_ntfs_spec.rb:31:in `with_acl_hash'
     # ./spec/integration/util/settings_ntfs_spec.rb:141:in `block (4 levels) in <top (required)>'
     # ./spec/integration/util/settings_ntfs_spec.rb:140:in `each'
     # ./spec/integration/util/settings_ntfs_spec.rb:140:in `block (3 levels) in <top (required)>'

Finished in 1 minute 10.98 seconds (files took 2.31 seconds to load)
2 examples, 0 failures, 1 pending
```

File: spec/integration/util/settings_ntfs_spec.rb [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      # Protected directories
+      protected_directory_settings.each do |setting|
+        with_acl_hash(Puppet.settings[setting]) do |acl|
+          expect(acl).not_to contain_any_inherited_aces
+        end
+      end
+    end
+
+    it 'should not add the NT AUTHORITY\\SYSTEM account, but should add BUILTIN\\Administrators' do
+      pending('PUP-6729 should resolve this')
```
[the remark this answers is not in the document]
Comment (Iristyle) [reply spec/integration/util/settings_ntfs_spec.rb]: As a follow on to this, some of the other work in PA-2112 / PA-2113 (and earlier PRs) resulted in perms getting set (inherited) as `SYSTEM: (F)` and `Administrators: (F)` across the `ProgramData\PuppetLabs` folder, which solves a number of the perms problems people have encountered in "the real world". That said, I think we still have an opportunity to make a few improvements to mode / perm handling. In particular, I don't think we should ever allows setting `SYSTEM` as anything other than `F`

**Comment by melissa** [comment]
@glennsarti I noticed you closed the associated ticket, citing that this work is no longer needed given the changes in the MSI. Can we close this PR as well?

**Comment by glennsarti** [comment]
@melissa Agreed. Closing

**Pull request closed without merging**